bin/reth/src/commands/import.rs

```diff
@@ -123,49 +132,55 @@ impl ImportCommand {
         let consensus = Arc::new(BeaconConsensus::new(self.chain.clone()));
         info!(target: "reth::cli", "Consensus engine initialized");
 
-        // create a new FileClient
-        info!(target: "reth::cli", "Importing chain file");
-        let file_client = Arc::new(FileClient::new(&self.path).await?);
-
-        // override the tip
-        let tip = file_client.tip().expect("file client has no tip");
-        info!(target: "reth::cli", "Chain file read");
-
-        let (mut pipeline, events) = self
-            .build_import_pipeline(
-                config,
-                provider_factory.clone(),
-                &consensus,
-                file_client,
-                StaticFileProducer::new(
+        // open file
+        let mut reader = ChunkedFileReader::new(&self.path, self.chunk_len).await?;
+
+        while let Some(file_client) = reader.next_chunk().await? {
+            // create a new FileClient from chunk read from file
+            info!(target: "reth::cli",
+                "Importing chain file chunk"
+            );
+
+            // override the tip
+            let tip = file_client.tip().expect("file client has no tip");
+            info!(target: "reth::cli", "Chain file chunk read");
+
+            let (mut pipeline, events) = self
+                .build_import_pipeline(
```

Comment:
we should be able to build the pipeline outside of the loop, same for the handle events below.

and then we just set the tip, and run it inside the loop

Reply:
but the pipeline gets a new `FileClient` every time, wdym?

I thought of moving the chunk reading code into the file client, and reading each chunk as part of a headers request, BUT it's not that simple. it would require adding a new cli arg with the tip, as it cannot be known from the file without reading the whole file. since this would require a non trivial amount of work, I left it for now, and thought of making it an issue to re-design. wdyt?